controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -409,7 +409,35 @@ func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs
 		return allErrs
 	}
 
-	// since upgrades to the next minor version are allowed, irrespective of the patch version
+	// Check if we're trying to upgrade to Kubernetes v1.19.0, which is not supported.
+	//
+	// In Kubernetes v1.19.0, the certificates API (which does CertificateSigningRequests
+	// for node bootstrapping, among other things) introduced version v1.
+	// The kube-controller-manager switched to only speaking v1 of this API.
+	//
+	// If you have a control plane with a load balancer in front, it is possible/likely that during an upgrade from 1.18 to 1.19,
+	// the 1.18 kube-controller-manager will be removed (as part of deleting one of the old machines),
+	// and the 1.19 kube-controller-manager will potentially acquire the leader election lock.
+	// It then connects to the load balancer, and there’s a chance it gets routed to one of the remaining 1.18 apiservers.
+	// If/when this happens, the kube-controller-manager will be unable to process any new CertificateSigningRequests
+	// (because it’s asking for certificates v1, which doesn’t exist in 1.18).
+	//
+	// This means all new node bootstrapping will fail.
+	//
+	// The solution is to update kubeadm so the kube-controller-manager, kube-scheduler,
+	// and kubelet of control plane nodes talk to the local API endpoint for the apiserver
+	// (i.e. the local apiserver static pod), instead of going through the load balancer.
+	if fromVersion.NE(toVersion) && toVersion.Equals(semver.MustParse("1.19.0")) {
```

Comment:
why not block 1.19.* ? in case v1.19.1 drops before we have time to update kubeadm

Reply:
@neolit123 Is already on it, it should come in v1.19.1. We can block the range although then we might have folks to upgrade

Reply:
👍